rdtools/normalization.py

```diff
@@ -209,175 +258,524 @@ def normalize_with_sapm(energy, sapm_kws):
-
-        insolation = irrad * irrad_tds
-        insolation = insolation.resample(freq).sum()
-        insolation = insolation.reindex(energy.index, method='nearest')
-
-    elif mean_model_td > mean_measure_td:
-        dc_power = dc_power.resample(freq).asfreq()
-        dc_power = dc_power.interpolate()
-        dc_power = dc_power.reindex(energy.index, method='nearest')
-        energy_dc = dc_power * measure_tds  # timedelta is that of measurment due to reindex
+    power_dc, irrad = sapm_dc_power(**sapm_kws)
 
-        irrad = irrad.resample(freq).asfreq()
-        irrad = irrad.interpolate()
-        irrad = irrad.reindex(energy.index, method='nearest')
-        insolation = irrad * measure_tds  # timedelta is that of measurment due to reindex
+    energy_normalized, insolation = normalize_with_expected_power(energy, power_dc, irrad, pv_input='energy')
 
-    normalized_energy = energy / energy_dc
-
-    return normalized_energy, insolation
+    return energy_normalized, insolation
 
 
 def delta_index(series):
     '''
-    Takes a panda series with a DatetimeIndex as input and
+    Takes a pandas series with a DatetimeIndex as input and
     returns (time step sizes, average time step size) in hours
+
+    Parameters
+    ----------
+    series : pd.Series
+        A pandas timeseries
+
+    Returns
+    -------
+    deltas : pd.Series
+        A timeseries representing the timestep sizes of `series`
+    mean : float
+        The average timestep
     '''
 
     if series.index.freq is None:
-        # If there is no frequency information, explicily calculate interval sizes
-        # Length of each interval calculated by using 'int64' to convert to nanoseconds
+        # If there is no frequency information, explicitly calculate interval
+        # sizes. Length of each interval calculated by using 'int64' to convert
+        # to nanoseconds.
         hours = pd.Series(series.index.astype('int64') / (10.0**9 * 3600.0))
         hours.index = series.index
         deltas = hours.diff()
     else:
-        # If there is frequency information, pandas shift can be used to gain a meaningful
-        # interful for the first element of the timeseries
-        # Length of each interval calculated by using 'int64' to convert to nanoseconds
-        deltas = (series.index - series.index.shift(-1)).astype('int64') / (10.0**9 * 3600.0)
+        # If there is frequency information, pandas shift can be used to gain
+        # a meaningful interval for the first element of the timeseries
+        # Length of each interval calculated by using 'int64' to convert to
+        # nanoseconds.
+        deltas = (series.index - series.index.shift(-1)).astype('int64') / \
+                    (10.0**9 * 3600.0)
     return deltas, np.mean(deltas.dropna())
 
 
-def irradiance_rescale(irrad, modeled_irrad, max_iterations=100, method=None):
+def irradiance_rescale(irrad, irrad_sim, max_iterations=100,
```

Comment:
This is a cool idea for a function. Is there a citation for this method? This seems like another great candidate for an example notebook.

Reply:
I don't believe we have a citation for this function. It's always good to add useful examples. I'm not sure we'll get to it before release, but I'll open an issue so we keep this on the radar. Thanks!

Reply:
This is the same idea as is used in `pvlib.clearsky.detect_clearsky`, to scale the modeled clearsky irradiance to maximize the number of detected clearsky points.  The piece of code that finds `fact` (`alpha` in the pvlib function) will be separated into its own function in [this PR](https://github.com/pvlib/pvlib-python/pull/1074/). It's pretty short code, though, maybe not worth the effort of re-using it.

Reply:
That's something to keep an eye on. It looks to me like both still use their respective definition of clear sky detection internally, so reusing at this stage would be tricky. But if either gets more generalized we could consider reuse for this.

Reply:
It's only [this short block of code](https://github.com/pvlib/pvlib-python/blob/3e25627e34bfd5aadea041da85a30626322b3a99/pvlib/clearsky.py#L763-L773) that will be pulled out into a helper function

Reply:
Ah I see, thank you for clarifying. Given the timelines involved, let's keep them separate for now, but keep this in mind for the future.